kratos/utilities/math_utils.h

```diff
@@ -702,14 +703,7 @@ class KRATOS_API(KRATOS_CORE) MathUtils
         const Vector& rSecondVector
         )
     {
-        Vector::const_iterator i = rFirstVector.begin();
-        Vector::const_iterator j = rSecondVector.begin();
-        double temp = 0.0;
-        while(i != rFirstVector.end()) {
-            temp += *i++ * *j++;
-        }
-        return temp;
-        //return std::inner_product(rFirstVector.begin(), rFirstVector.end(), rSecondVector.begin(), 0.0);
+        return std::inner_product(rFirstVector.begin(), rFirstVector.end(), rSecondVector.cbegin(), 0.0);
```

Comment:
Just to avoid potential performance degradation / confusion in case of someone refactoring this function or copying it elsewhere, cast the initial value to the same type as the iterator is pointing to.

Also, the arguments are both `const` so no need to explicitly invoke `Vector::cbegin`.

```suggestion
        return std::inner_product(rFirstVector.begin(), rFirstVector.end(), rSecondVector.begin(), static_cast<Vector::value_type>(0.0));
```